Traceback (most recent call last):
  File "/Users/kgoebber/miniconda3/envs/metpy_book/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/kgoebber/miniconda3/envs/metpy_book/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kgoebber/miniconda3/envs/metpy_book/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kgoebber/miniconda3/envs/metpy_book/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/kgoebber/miniconda3/envs/metpy_book/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/kgoebber/miniconda3/envs/metpy_book/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/kgoebber/miniconda3/envs/metpy_book/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Set date that you want
# Data goes back to the 1970's
date = datetime(2007, 5, 4, 18)

# Set station ID, there are different stations back in the day
# Current station IDs found at http://weather.rap.ucar.edu/upper
station = 'DDC'

# Use Siphon module to grab data from remote server
df = WyomingUpperAir.request_data(date, station)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mHTTPError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[2], line 10[0m
[1;32m      7[0m station [38;5;241m=[39m [38;5;124m'[39m[38;5;124mDDC[39m[38;5;124m'[39m
[1;32m      9[0m [38;5;66;03m# Use Siphon module to grab data from remote server[39;00m
[0;32m---> 10[0m df [38;5;241m=[39m [43mWyomingUpperAir[49m[38;5;241;43m.[39;49m[43mrequest_data[49m[43m([49m[43mdate[49m[43m,[49m[43m [49m[43mstation[49m[43m)[49m

File [0;32m~/miniconda3/envs/metpy_book/lib/python3.11/site-packages/siphon/simplewebservice/wyoming.py:50[0m, in [0;36mWyomingUpperAir.request_data[0;34m(cls, time, site_id, **kwargs)[0m
[1;32m     30[0m [38;5;250m[39m[38;5;124mr[39m[38;5;124;03m"""Retrieve upper air observations from the Wyoming archive.[39;00m
[1;32m     31[0m 
[1;32m     32[0m [38;5;124;03mParameters[39;00m
[0;32m   (...)[0m
[1;32m     47[0m 
[1;32m     48[0m [38;5;124;03m"""[39;00m
[1;32m     49[0m endpoint [38;5;241m=[39m [38;5;28mcls[39m()
[0;32m---> 50[0m df [38;5;241m=[39m [43mendpoint[49m[38;5;241;43m.[39;49m[43m_get_data[49m[43m([49m[43mtime[49m[43m,[49m[43m [49m[43msite_id[49m[43m)[49m
[1;32m     51[0m [38;5;28;01mreturn[39;00m df

File [0;32m~/miniconda3/envs/metpy_book/lib/python3.11/site-packages/siphon/simplewebservice/wyoming.py:70[0m, in [0;36mWyomingUpperAir._get_data[0;34m(self, time, site_id)[0m
[1;32m     53[0m [38;5;28;01mdef[39;00m [38;5;21m_get_data[39m([38;5;28mself[39m, time, site_id):
[1;32m     54[0m [38;5;250m    [39m[38;5;124mr[39m[38;5;124;03m"""Download and parse upper air observations from an online archive.[39;00m
[1;32m     55[0m 
[1;32m     56[0m [38;5;124;03m    Parameters[39;00m
[0;32m   (...)[0m
[1;32m     68[0m 
[1;32m     69[0m [38;5;124;03m    """[39;00m
[0;32m---> 70[0m     raw_data [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_get_data_raw[49m[43m([49m[43mtime[49m[43m,[49m[43m [49m[43msite_id[49m[43m)[49m
[1;32m     71[0m     soup [38;5;241m=[39m BeautifulSoup(raw_data, [38;5;124m'[39m[38;5;124mhtml.parser[39m[38;5;124m'[39m)
[1;32m     72[0m     tabular_data [38;5;241m=[39m StringIO(soup[38;5;241m.[39mfind_all([38;5;124m'[39m[38;5;124mpre[39m[38;5;124m'[39m)[[38;5;241m0[39m][38;5;241m.[39mcontents[[38;5;241m0[39m])

File [0;32m~/miniconda3/envs/metpy_book/lib/python3.11/site-packages/siphon/simplewebservice/wyoming.py:144[0m, in [0;36mWyomingUpperAir._get_data_raw[0;34m(self, time, site_id)[0m
[1;32m    126[0m [38;5;250m[39m[38;5;124;03m"""Download data from the University of Wyoming's upper air archive.[39;00m
[1;32m    127[0m 
[1;32m    128[0m [38;5;124;03mParameters[39;00m
[0;32m   (...)[0m
[1;32m    138[0m 
[1;32m    139[0m [38;5;124;03m"""[39;00m
[1;32m    140[0m path [38;5;241m=[39m ([38;5;124m'[39m[38;5;124m?region=naconf&TYPE=TEXT[39m[38;5;124m%[39m[38;5;124m3ALIST[39m[38;5;124m'[39m
[1;32m    141[0m         [38;5;124m'[39m[38;5;124m&YEAR=[39m[38;5;124m{[39m[38;5;124mtime:[39m[38;5;124m%[39m[38;5;124mY}&MONTH=[39m[38;5;124m{[39m[38;5;124mtime:[39m[38;5;124m%[39m[38;5;124mm}&FROM=[39m[38;5;124m{[39m[38;5;124mtime:[39m[38;5;132;01m%d[39;00m[38;5;124m%[39m[38;5;124mH}&TO=[39m[38;5;124m{[39m[38;5;124mtime:[39m[38;5;132;01m%d[39;00m[38;5;124m%[39m[38;5;124mH}[39m[38;5;124m'[39m
[1;32m    142[0m         [38;5;124m'[39m[38;5;124m&STNM=[39m[38;5;132;01m{stid}[39;00m[38;5;124m'[39m)[38;5;241m.[39mformat(time[38;5;241m=[39mtime, stid[38;5;241m=[39msite_id)
[0;32m--> 144[0m resp [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mget_path[49m[43m([49m[43mpath[49m[43m)[49m
[1;32m    145[0m [38;5;66;03m# See if the return is valid, but has no data[39;00m
[1;32m    146[0m [38;5;28;01mif[39;00m resp[38;5;241m.[39mtext[38;5;241m.[39mfind([38;5;124m"[39m[38;5;124mCan[39m[38;5;124m'[39m[38;5;124mt[39m[38;5;124m"[39m) [38;5;241m!=[39m [38;5;241m-[39m[38;5;241m1[39m:

File [0;32m~/miniconda3/envs/metpy_book/lib/python3.11/site-packages/siphon/http_util.py:457[0m, in [0;36mHTTPEndPoint.get_path[0;34m(self, path, query)[0m
[1;32m    434[0m [38;5;28;01mdef[39;00m [38;5;21mget_path[39m([38;5;28mself[39m, path, query[38;5;241m=[39m[38;5;28;01mNone[39;00m):
[1;32m    435[0m [38;5;250m    [39m[38;5;124;03m"""Make a GET request, optionally including a query, to a relative path.[39;00m
[1;32m    436[0m 
[1;32m    437[0m [38;5;124;03m    The path of the request includes a path on top of the base URL[39;00m
[0;32m   (...)[0m
[1;32m    455[0m 
[1;32m    456[0m [38;5;124;03m    """[39;00m
[0;32m--> 457[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mget[49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43murl_path[49m[43m([49m[43mpath[49m[43m)[49m[43m,[49m[43m [49m[43mquery[49m[43m)[49m

File [0;32m~/miniconda3/envs/metpy_book/lib/python3.11/site-packages/siphon/http_util.py:492[0m, in [0;36mHTTPEndPoint.get[0;34m(self, path, params)[0m
[1;32m    490[0m     [38;5;28;01melse[39;00m:
[1;32m    491[0m         text [38;5;241m=[39m resp[38;5;241m.[39mtext
[0;32m--> 492[0m     [38;5;28;01mraise[39;00m requests[38;5;241m.[39mHTTPError([38;5;124m'[39m[38;5;124mError accessing [39m[38;5;132;01m{0}[39;00m[38;5;130;01m\n[39;00m[38;5;124m'[39m
[1;32m    493[0m                              [38;5;124m'[39m[38;5;124mServer Error ([39m[38;5;132;01m{1:d}[39;00m[38;5;124m: [39m[38;5;132;01m{2}[39;00m[38;5;124m)[39m[38;5;124m'[39m[38;5;241m.[39mformat(resp[38;5;241m.[39mrequest[38;5;241m.[39murl,
[1;32m    494[0m                                                                 resp[38;5;241m.[39mstatus_code,
[1;32m    495[0m                                                                 text))
[1;32m    496[0m [38;5;28;01mreturn[39;00m resp

[0;31mHTTPError[0m: Error accessing http://weather.uwyo.edu/cgi-bin/sounding/?region=naconf&TYPE=TEXT%3ALIST&YEAR=2007&MONTH=05&FROM=0418&TO=0418&STNM=DDC
Server Error (503: Sorry, the server is too busy to process your request.
Please try again later.
)

